### ISSUE TITLE
Get Python prompts from Python

### ISSUE THREAD

**Issue by rob-smallshire** [sixty-north/gjenta#41]
Rather than hard-coding the various Python prompts into Python, we get at least some of them by asking Python for them.

In particular the `>>>` is available as `sys.ps1` and `...` is available as `sys.ps2`.

The will likely will help with the fact that Gjenta doesn't currently respect customised prompts, for example when the prompt has been coloured using control codes.

https://github.com/sixty-north/gjenta/blob/b605b3ec391d45dd8a2d18ec53ab2030ad5ebdf3/src/gjenta/interpreters.py#L48

**Comment by abingham** [comment]
This is not entirely straightforward. The way pexpect works is that you need to know your prompts beforehand; it needs to know them in order to know how to process the tty. So I think we'll need to probe Python beforehand, for example by starting a short-lived Python repl to probe it for its prompts.

The obvious approach doesn't seem to work:
```
% python -c "import sys; print(sys.ps1)"                                                                                                                                                                                                                                                       lib/python3.7 goldenhammer
Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: module 'sys' has no attribute 'ps1'
```
It seem that `ps1` isn't initialized in this case. Perhaps there's a way to poke the runtime to initialize it.

**Comment by abingham** [comment]
It seems you can force the initialization of `sys.ps1` and `sys.ps2` by starting Python with `-i`:
```
% echo "import sys; print(sys.ps1)" | python -i                                                                                                                                                                                                                                                                                    repos/cpython (master) goldenhammer
Python 3.7.0 (default, Mar 13 2019, 07:59:20)
[Clang 10.0.0 (clang-1000.10.44.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> >>>
>>>
 % echo "import sys; print(sys.ps2)" | python -i                                                                                                                                                                                                                                                                                    repos/cpython (master) goldenhammer
Python 3.7.0 (default, Mar 13 2019, 07:59:20)
[Clang 10.0.0 (clang-1000.10.44.4)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> ...
>>>
```
So we *can* probe for these values. We just need a way of reliably parsing the values out.

**Comment by abingham** [comment]
This is addressed in 830f0a2ec7a4fd90b567bc8614441fbe9434bcd3
